src/response.rs

```diff
@@ -0,0 +1,59 @@
+use core::fmt::{Debug, Write};
+use serde::{Deserialize, Serialize};
+
+use heapless::{String, Vec};
+
+// The maximum size of a response.
+const MAX_RESPONSE_SIZE: usize = 128;
+
+/// Responses are always generated as a result of handling an in-bound request.
+#[derive(Debug, Default, Clone, Deserialize, Serialize)]
+pub struct Response {
+    pub code: i32,
+    pub data: Vec<u8, MAX_RESPONSE_SIZE>,
+}
+
+impl Response {
+    /// A response without data indicating success.
+    pub fn ok() -> Self {
+        Self::custom(0, "Ok")
+    }
+
+    /// A response indicating failure with some error code.
+    pub fn error(err: impl Debug) -> Self {
```

Comment:
Rather `impl Display`?
`Debug` seems a bit code oriented.

Reply:
The main goal here is to be able to provide `Error::<Variant>` enums to this function so they can be stringified. `Display` is often not implemented for many embedded no_std types, so I went for `Debug` here instead since I figured it would be more versatile.

Reply:
The trait to do user-facing stringification is `Display`.
But I agree that `Debug` is more common as it's derivable.